### PR TITLE
[2.0] SDL bindings

### PR DESCRIPTION
**Pull request by Perksey**
# Summary of the PR
Adds SDL bindings.

# Related issues, Discord discussions, or proposals
#211 
#48 

# Further Comments
This is a precursor to a subsequent PR containing a SDL windowing backend.

### CODE REVIEW

**Comment by Perksey** [comment]
Ready for review. Notes for review:
- There are some anonymous structs I haven't had the chance to give names yet. Ideally they should get names at some point during the preview cycle, but they're luckily only for one of the lesser-used headers of SDL.
- Some binder changes were made.
- The rest of the PR is pretty much just generated code so it should be fairly easy to review.

File: generator.json [hunk 1]
```diff
@@ -342,7 +342,8 @@
                     "PFN_vkVoidFunction": "FuncPtr",
                     "PFN_vkDebugReportCallbackEXT": "FuncPtr",
                     "PFN_vkDebugUtilsMessengerCallbackEXT": "FuncPtr",
-                    "char": "byte"
+                    "char": "byte",
+                    "Window": "IntPtr"
```
[the remark this answers is not in the document]
Comment (Perksey) [reply generator.json:346]: That's not valid JSON.

File: src/Core/Silk.NET.BuildTools/Common/Functions/Parameter.cs [hunk 2]
```diff
@@ -47,37 +47,37 @@ public class Parameter : IEquatable<Parameter>
...
+        // public bool Equals(Parameter other)
+        // {
+        //     if (ReferenceEquals(null, other))
+        //     {
+        //         return false;
+        //     }
+// 
+        //     if (ReferenceEquals(this, other))
+        //     {
+        //         return true;
```
[the remark this answers is not in the document]
Comment (Perksey) [reply src/Core/Silk.NET.BuildTools/Common/Functions/Parameter.cs:60]: Commented out unused equality members that only made JSON serialization hell.

[the remark this answers is not in the document]
Comment (HurricanKai) [reply src/Core/Silk.NET.BuildTools/Common/Functions/Parameter.cs:60]: I'd just remove them

File: generator.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (HurricanKai) [reply generator.json:346]: Not technically, but all parsers have an option to allow that. I'd set it and add the extra `,` but out of scope of this PR, understandably

File: src/Core/Silk.NET.BuildTools/Cpp/Clang.cs [hunk 3]
```diff
@@ -34,9 +32,15 @@ public static unsafe Profile GenerateProfile(string fileName, Stream input, Bind
                 Name = Path.GetFileNameWithoutExtension(fileName)
             };
 
-            var traversals = task.ClangOpts.Traverse.Select
-                    (x => Path.GetFullPath(x).ToLower().Replace('\\', '/'))
+            //var traversals = task.ClangOpts.Traverse.Select(x => Path.GetFullPath(x).ToLower().Replace('\\', '/'))
+            //    .ToArray();
```
[the remark this answers is not in the document]
Comment (Perksey) [reply src/Core/Silk.NET.BuildTools/Cpp/Clang.cs]: Removed what I'm willing to remove - I want to keep the commented out code in VisitDecl so I know how much of the AST I've covered.

File: src/Core/Silk.NET.BuildTools/Common/Functions/Parameter.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Perksey) [reply src/Core/Silk.NET.BuildTools/Common/Functions/Parameter.cs]: Done

**Comment by Perksey** [comment]
Ready for re-review.

**Pull request merged**